packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -1012,7 +1049,7 @@ export const Table = <
                           }
                         }}
                       >
-                        <ShadowRoot>
+                        <ShadowRoot key={`${columnProps.key}_shadow_root`}>
```

Comment:
Are you sure you need to key all the children separately? Usually keys are only required for lists of repeated elements of the same type.

Reply:
Yes, the children need to be keyed separately to resolve the react key warnings, since the children are rendered as siblings of the `ShadowRoot` component in the react component tree. 
![image](https://github.com/iTwin/iTwinUI/assets/141063311/06bc851b-4327-468e-8d35-5e73aa334032)
I do wonder why react thinks this list needs to be keyed, though. The components are of different types, even if they are siblings.

Reply:
It looks like using a fragment would avoid the need for separate keys.

![](https://github.com/iTwin/iTwinUI/assets/9084735/d2ae86cc-c182-4a25-bf25-8fee743cc3f3)

Reply:
Wrapped the `ShadowRoot` and its children in fragment and removed their keys in this [commit](https://github.com/iTwin/iTwinUI/pull/2092/commits/de6dbda7a39f0048763514b26b0521d8b7a085a4).